lang/en/local_sitenotice.php

```diff
@@ -55,6 +55,8 @@
 $string['notice:reqack'] = 'Requires acknowledgement';
 $string['notice:reqack_help'] = 'If enabled, the user will need to accept the notice before they can continue to use the LMS site.
 If the user does not accept the notice, he/she will be logged out of the site.';
+$string['notice:forcelogout'] = 'Force logout';
+$string['notice:forcelogout_help'] = 'If enabled, the user will be logged out after closing the notice. CAUTION: Do NOT enable this setting if the notice audience is set to "All users", doing so can potentially lock all users out of the site. ';
```

Comment:
let's exclude exclude admins from logging out as we discussed?

Reply:
Ohhh sorry, I misunderstood your original comment. Will add that functionality.

Reply:
Added. Site admins are now excluded from the reqack setting and the forcelogout setting. Happy to change this to just force logout if the reqack setting does need to be honoured even for site admins.

Reply:
let's leave reqack logic as is. So admins have to acknowledge before to proceed. But not for forcelogout.

Reply:
I've removed the function `optionally_logout` and left the logic in the individual functions. I've created #46 to capture the refactor work in the future.